packages/@headlessui-react/src/components/popover/popover.tsx

```diff
@@ -360,61 +206,46 @@ function PopoverFn<TTag extends ElementType = typeof DEFAULT_POPOVER_TAG>(
     (event) => {
       if (event.target === window) return
       if (!DOM.isHTMLorSVGElement(event.target)) return
-      if (popoverState !== PopoverStates.Open) return
+      if (machine.state.popoverState !== PopoverStates.Open) return
```

Comment:
Am curious why this  needed to change? Or was it just for consistency elsewhere maybe?

Reply:
Yeah just consistency because you mentioned it in another PR hehe (https://github.com/tailwindlabs/headlessui/pull/3685#discussion_r2037984939) and to have the latest and greatest state.